Stats revamp v2 views and visitors graph improvements

ravishanker opened the pull request:
This PR fixes all the outstanding issues on Views and Visitors graph from the following PRs
- [Views and Visitors Card](https://github.com/wordpress-mobile/WordPress-Android/pull/16258)
- [Views and Visitors Graph](https://github.com/wordpress-mobile/WordPress-Android/pull/16327)

Issues Fixed:
- Tool tip (Marker) now morphs to top, bottom, left and right based on location of the data point within graph
- Tool tip only appears only for current week line data
- data point circle only appears along with marker
- Shows ticks for data points on xAxis
- dates on xAxis are formatted better
- Copy of information line is updated from "Your this week's views ..". to "Your views this week"
- Chips are updated to correct colour
- Refactor and cleaned up

Fixes #16274 

| Marker | Marker | Marker
|---|---|---|
| ![Screenshot_20220517_001839](https://user-images.githubusercontent.com/990349/168614649-0b60002c-febc-44ab-826e-c24dd35cde19.png)|![Screenshot_20220517_001734](https://user-images.githubusercontent.com/990349/168614770-25a98792-e64f-4e1e-91ac-41317ddd7ad7.png)| ![Screenshot_20220517_001753](https://user-images.githubusercontent.com/990349/168614846-21afe92d-3d2e-47ef-8caf-e2d3ce300760.png)|


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab as in the images above (scroll down if necessary)

Test 2:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Notice that the Views graph is in blue and this week and previous week and percent change are all in order
- Notice the graph, data and change are all compare this week with previous week
- Switch to Visitors by tapping chip and notice graph is in purple color

Test 3:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Tap on the graph.  Notice a marker popup as in the third image above.


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Refactor date format

ravishanker committed:
Rename functions and params

ravishanker committed:
Improved MarkerView positioning

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Views-and-Visitors-Graph-Improvements

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Views-and-Visitors-Graph-Improvements

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Views-and-Visitors-Graph-Improvements

ravishanker committed:
Format xAxis

add ticks to xAxis
fix date labels

ravishanker committed:
Inject StatsUtils with Hilt

ravishanker committed:
update too tip to be adaptive

Make marker view adaptive to the bounds of graph

Update LineChartViewHolder.kt

ravishanker committed:
Update copy

ravishanker committed:
Update chip colors

irfano commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/LineChartLabelFormatter.kt:
```diff
@@ -1,17 +1,49 @@
 package org.wordpress.android.ui.stats.refresh.utils
 
+import android.nfc.FormatException
```
`nfc.FormatException` is not right I guess.

ravishanker replied:
Good catch!  Imports sometimes go wrong.  Fixed
Also fixed left rounded corner and accessibility crash

irfano reviewed (changes requested):
This looks very good! Tooltip position works great!

- Added one small comment about `android.nfc.FormatException`
- Upper left corner of the tooltip is not rounded. 
- Dates are null on my emulator. It could be a problem with my local. We discussed that before. I'll work I'll work on that
<img src="https://user-images.githubusercontent.com/2471769/169727396-fca57c74-61cc-4125-b616-66ce25e39227.png" width=320>

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Views-and-Visitors-Graph-Improvements

ravishanker committed:
Update LineChartLabelFormatter.kt

- Removed incorrect FormatException import
- Removed redundant try catch block from formatLabelDate method

ravishanker committed:
Update LineChartAccessibilityHelper.kt

fix crash in helper

ravishanker committed:
Update LineChartMarkerView.kt

fix the rounded corners on markerview

irfano reviewed (approved):
LGTM! 👍🏻 
Dates are null in my emulator. I'm adding this bug to the bug task in the project.

My only concern is accessibility. Talkback is not speaking anything when the line chart is selected.. Because I can't select any node on the line chart when talkback is active.
In the bar chart, I can select any day. Should we make improvements here, in this PR or a new PR, @ravishanker ? We can merge after deciding this.
